### DISCUSSION TITLE
Disable quantization during CLI mesh optimizing

### DISCUSSION THREAD

**Discussion by bolatuly** [donmccurdy/glTF-Transform#1523] [Q&A] [answered]
Hi! mesh optimizing uses `KHR_mesh_quantization` extension which is not supported by application I am using.
Is it possible to disable quantization during gltf-transform cli function `meshopt`?

To be able to do something like this:

```sh
gltf-transform meshopt input.glb output.glb --no-quantization
```

something like gltfpack tool which has option -noq to disable `KHR_mesh_quantization`

**Comment by donmccurdy** [comment]
@bolatuly could you share which flags you're setting on gltfpack, if that's providing results you prefer? If only `-noq` is specified then gltfpack skips meshopt compression entirely, and just does general optimization on the model. In glTF Transform that would be comparable to:

```
gltf-transform optimize in.glb out.glb --no-compress --no-instance
```

If you're also using the `-c` or `-cc` flags in gltfpack, then this does the same general optimization, then also applies meshopt compression, which would require your application to support `EXT_meshopt_compression`. In my experience it's unusual for an application to support `EXT_meshopt_compression` without also supporting `KHR_mesh_quantization`; they work best when used together. gltfpack offers the option to apply meshopt compression withou quantization, but glTF Transform currently does not. If that's an issue, I'd be interested to hear more about the use case and/or application if possible!

**Reply by bolatuly** [reply]
I use both `-c` and `-noq` options for gltfpack. Application used by me for model render is [Mapbox GL JS](https://github.com/mapbox/mapbox-gl-js/), they don't disclose much about the specific extensions they support for glb models. However, when I try to render a model with `EXT_meshopt_compression`, it works fine. But if I add `KHR_mesh_quantization`, it renders with some artefacts. I looked through their code and only found references to `EXT_meshopt_compression` support.

**Reply by donmccurdy** [reply, marked as answer]
I see, thanks! You're right, they support `EXT_meshopt_compression` but not `KHR_mesh_quantization`. It might be worth filing an issue for them about that, it's usually one of the simplest glTF extensions to import, but regardless. I've opened a new feature request about supporting meshopt compression without quantization. I can't promise any timelines for this, but PRs are always welcome.

- https://github.com/donmccurdy/glTF-Transform/issues/1524

In the meantime, Draco compression might be the next best option.

**Reply by bolatuly** [reply]
thanks for your reply! Originally I used Draco compression, but I liked the decoding speed of meshopt compression, so I started trying it out.

**Reply by donmccurdy** [reply]
In progress:

- https://github.com/donmccurdy/glTF-Transform/pull/1526
